src/views/routing.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect } from "react";
```

Comment:
I have a small preference to move this up a level so that `views` only contains the actual views and not the routing logic but I wouldn't want to block this on that. We can also revisit once the rest of the URL routing changes go in which would remove a chunk of this code as well.